specification/communication/data-plane/Chat/preview/2025-11-15-preview/communicationserviceschat.json

```diff
@@ -0,0 +1,2210 @@
+          },
+          {
+            "in": "body",
+            "name": "sendTypingNotificationRequest",
+            "description": "Details of the typing notification request.",
+            "schema": {
+              "$ref": "#/definitions/SendTypingNotificationRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful."
+          },
+          "401": {
+            "description": "Unauthorized.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "403": {
+            "description": "Forbidden.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "429": {
+            "description": "Too many requests.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
+          "503": {
+            "description": "Service unavailable.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          }
+        },
+        "x-ms-examples": {
+          "Post typing event to a thread": {
+            "$ref": "./examples/Messages_SendTypingNotification.json"
+          }
+        }
+      }
+    },
+    "/chat/threads/{chatThreadId}/images": {
+      "post": {
+        "tags": [
+          "Images"
+        ],
+        "summary": "Upload an image in a thread, on behalf of a user.",
+        "operationId": "ChatThread_UploadChatImage",
+        "consumes": [
+          "application/octet-stream"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "header",
+            "name": "image-filename",
+            "description": "The file name of the image.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "in": "path",
+            "name": "chatThreadId",
+            "description": "Thread id where the uploaded image belongs to. (Teams meeting only)",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "chatImageFile",
+            "description": "Image binary data, allowed image formats: jpeg, png, gif, heic, webp",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ChatImageFile"
+            }
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Uploaded successfully, the `Location` header contains the URL for the newly uploaded image.",
```

Comment:
Should you define the Location header in the API?

Reply:
Haven't found any reference in other specs where we're able to include header information for responses though, by default the response schema is for the body. We do include header information in the examples though.